src/components/intent.ts

```diff
@@ -625,6 +625,25 @@ export class Intent {
         return this.client.setAvatarUrl(url);
     }
 
+    /**
+     * Ensure that the user has the given profile information set. If it does not,
+     * set it.
+     * @param displayname The displayname to set. Leave undefined to ignore.
+     * @param avatarUrl The avatar to set. Leave undefined to ignore.
+     */
+    public async ensureProfile(displayname?: string, avatarUrl?: string) {
```

Comment:
Might be a bit too much of an edge case:
How to ensure the display name or avatarUrl is unset?

Falsy values will get ignored or cause the function to throw.

Reply:
Hmm, good question. In fairness the spec doesn't actually give you a way to unset a profile https://spec.matrix.org/unstable/client-server-api/#profiles so it's always been a bit troublesome. I'd probably say for now that the bridge should explicitly unset a profile key if they want to remove it rather than calling this function.